http4k-server/jetty11/src/main/kotlin/org/http4k/server/Http4kWebSocketFrameHandler.kt

```diff
@@ -1,37 +1,44 @@
 package org.http4k.server
 
+import org.eclipse.jetty.util.BufferUtil
+import org.eclipse.jetty.util.ByteArrayOutputStream2
 import org.eclipse.jetty.util.Callback
-import org.eclipse.jetty.util.Utf8StringBuilder
 import org.eclipse.jetty.websocket.core.CloseStatus
 import org.eclipse.jetty.websocket.core.CoreSession
 import org.eclipse.jetty.websocket.core.Frame
 import org.eclipse.jetty.websocket.core.FrameHandler
 import org.eclipse.jetty.websocket.core.OpCode.BINARY
 import org.eclipse.jetty.websocket.core.OpCode.CONTINUATION
 import org.eclipse.jetty.websocket.core.OpCode.TEXT
-import org.http4k.core.Body
-import org.http4k.core.StreamBody
+import org.http4k.core.MemoryBody
 import org.http4k.websocket.PushPullAdaptingWebSocket
 import org.http4k.websocket.WsConsumer
 import org.http4k.websocket.WsMessage
 import org.http4k.websocket.WsStatus
 
 class Http4kWebSocketFrameHandler(private val consumer: WsConsumer) : FrameHandler {
     private var websocket: PushPullAdaptingWebSocket? = null
-    private val textBuffer = Utf8StringBuilder()
+    private val buffer = ByteArrayOutputStream2()
+    private var messageMode =  WsMessage.Mode.Text
 
     override fun onFrame(frame: Frame, callback: Callback) {
         try {
             when (frame.opCode) {
+                TEXT -> messageMode = WsMessage.Mode.Text
```

Comment:
Since none of our clients and servers can agree on a frame size, I'm unable to test `CONTINUATION` frames generically.  I believe `CONTINUATION` frames must always be preceeded by a `DATA` or `TEXT` frame, but I have no idea how this will work with parallel requests.

Reply:
Seems like a reasonable approach 😄